js/src/as-report.js

```diff
@@ -4,45 +4,60 @@
  */
 import { __ } from '@wordpress/i18n';
 import apiFetch from '@wordpress/api-fetch';
-import { Component } from '@wordpress/element';
-import { Fragment } from '@wordpress/element';
-import { Card, EmptyContent, ReportFilters, TableCard, TablePlaceholder } from '@woocommerce/components';
+import { Component, Fragment } from '@wordpress/element';
 import { map } from 'lodash';
 
+/**
+ * WooCommerce dependencies
+ */
+import { Card, EmptyContent, ReportFilters, TableCard, TablePlaceholder } from '@woocommerce/components';
+import { onQueryChange, stringifyQuery } from '@woocommerce/navigation';
+
 /**
  * Internal dependencies
  */
+import { statusFilters } from './config';
 const showDatePicker = false;
 
 //@todo: import './style.scss'; if necessary
 
 class ActionsReport extends Component {
 
-	constructor() {
-		super();
+	constructor( props ) {
+		super( props );
 		this.state = {
 			actions: null,
 			loading: true,
 		};
+		this.getHeadersContent = this.getHeadersContent.bind( this );
+		this.getRowsContent = this.getRowsContent.bind( this );
 	}
 
 	componentDidMount() {
 		// This should be handled for us automagically, but the nonce wasn't working for me
 		apiFetch.use( apiFetch.createNonceMiddleware( asaSettings.nonce ) );
 		apiFetch.use( apiFetch.createRootURLMiddleware( asaSettings.api_root ) );
-// @todo: add query parameters
-		this.fetchActionData();
+
+		this.fetchActionData( {
+			orderby: 'scheduled',
+			order: 'asc',
+			status: 'pending',
```

Comment:
You should grab this from the url. If I refresh a page with `?status=complete`, then `pending` gets sent in the request.

Reply:
I added this. Compared to the other WCA components it will seem a bit weird that you need to add `?status=all` to get the list with all statuses.